DOCS: add user guide page for Desktop session behavior (#7800)
ansys/pyaedt: doc/source/User_guide/desktop_sessions.rst

Desktop sessions
================

PyAEDT can use ``Desktop`` either to start a new AEDT session or to connect to an
existing one. The ``close_on_exit`` argument controls whether that AEDT session is
closed when leaving a context manager or when the Python interpreter session ends.

When ``close_on_exit`` is not explicitly provided, PyAEDT chooses the behavior automatically:

- Inside a context manager (``with Desktop(...)``), ``close_on_exit`` is treated as ``True``.
- Outside a context manager:

>   - if PyAEDT starts a new AEDT session, ``close_on_exit`` behaves as ``True``
>   - if PyAEDT connects to an existing AEDT session, ``close_on_exit`` behaves as ``False``

Reviewer: ```suggestion
  - if PyAEDT starts a new AEDT session, ``close_on_exit`` behaves as ``True``,
  - if PyAEDT connects to an existing AEDT session, ``close_on_exit`` behaves as ``False``.
```

Reviewer 2: done

When ``close_on_exit`` is explicitly set to ``True`` or ``False``, the user choice is always respected.

Behavior summary
----------------

The following table summarizes the most common behavior combinations:

.. list-table::
   :header-rows: 1
   :widths: 25 20 20 35

>    * - Usage

Reviewer: I think part of this table should be removed or reworked. Here is the current content

<img width="1246" height="636" alt="Image" src="https://github.com/user-attachments/assets/7719ae24-f476-4818-af7c-9e73e2c43329" />

and I find the next section redundant (but I think it is more understandable than the table).

Reviewer 2: The text version with bullet points was already included in the docstring. The request was to create the table to complement it.

Reviewer: I understand that but I find that the table is difficult to understand as it is. More over, there is a lot of redundancy into it. Take the `with ...` section as an example, there is only one aspect that is of importance (the `close_on_exit` input), and I think that two lines could be merged (`omitted` or `True`)

   * - ``with Desktop(...)``
     - New or existing session
     - ``True``
     - AEDT closes when leaving the context manager
   * - ``with Desktop(...)``
     - New or existing session
     - ``False``
     - AEDT remains open when leaving the context manager
   * - ``Desktop(...)``
     - New session started by PyAEDT
     - omitted
     - AEDT closes when the interpreter session ends
   * - ``Desktop(...)``
     - PyAEDT connected to an existing session
     - omitted
     - AEDT remains open when the interpreter session ends
   * - ``Desktop(...)``
     - New or existing session
     - ``True``
     - AEDT closes when the interpreter session ends
   * - ``Desktop(...)``
     - New or existing session
     - ``False``
     - AEDT remains open when the interpreter session ends

Use a context manager
---------------------

Use a context manager when you want deterministic cleanup at the end of a block:

.. code:: python

    import ansys.aedt.core

    with ansys.aedt.core.Desktop(
        version="2026.1",
        non_graphical=True,
        new_desktop=True,
    ):
        hfss = ansys.aedt.core.Hfss()
        maxwell = ansys.aedt.core.Maxwell3d()

>         # Work with AEDT here.

Reviewer: ```suggestion
        # ...
        # Work with AEDT here.
        # ...
```
Very subjective but I like it better that way than just with the direct sentence. If agreed, this should be applied in other locations.

Reviewer 2: done

If needed, you can still override the default behavior explicitly:

.. code:: python

    import ansys.aedt.core

    with ansys.aedt.core.Desktop(
        version="2026.1",
        non_graphical=True,
        new_desktop=False,
        close_on_exit=False,
    ):
        hfss = ansys.aedt.core.Hfss()
        maxwell = ansys.aedt.core.Maxwell3d()
        # Work with an existing AEDT session here.

    # The AEDT session remains open here.

Use ``Desktop`` class directly
--------------------

When ``Desktop`` is used directly, the default behavior depends on whether PyAEDT starts
or attaches to AEDT. You can also release the desktop explicitly for finer control:

.. code:: python

    import ansys.aedt.core

    d = ansys.aedt.core.Desktop(
        version="2026.1",
        non_graphical=False,
        new_desktop=False,
    )

    hfss = ansys.aedt.core.Hfss()
    # Work with AEDT here.

    d.release_desktop(close_projects=False, close_desktop=False)

Recommendations
---------------

- Use ``with Desktop(...)`` when you want predictable cleanup.
- Use direct ``Desktop(...)`` construction when you need more manual control over the AEDT session lifecycle.

> - When attaching to an existing AEDT session, consider leaving ``close_on_exit`` unset or setting it explicitly to ``False`` if you do not want PyAEDT to close that session.

Reviewer: ```suggestion
- When attaching to an existing AEDT session, consider leaving ``close_on_exit`` unset or setting it explicitly to ``True`` if you do want PyAEDT to close that session.
```
Isn't the default behavior not to close the session in that case ? If that's the case then leaving it is the safe choice and only setting it to `True` when we want to close the session should be explained here.
